spec/support/vendor_api/shared_examples.rb

```diff
@@ -0,0 +1,21 @@
+RSpec.shared_examples 'an endpoint that requires metadata' do |action|
+  it 'returns an error when Metadata is not provided' do
+    application_choice = create(:application_choice)
+
+    post "/api/v1/applications/#{application_choice.id}/#{action}"
+
+    expect(response).to have_http_status(422)
+    expect(parsed_response).to be_valid_against_openapi_schema('BadRequestBodyResponse')
+  end
+
+  it 'returns an error when Metadata is invalid' do
+    application_choice = create(:application_choice)
+
+    invalid_metadata = { invalid: :metadata }
+
+    post "/api/v1/applications/#{application_choice.id}/#{action}", params: { meta: invalid_metadata }
+
+    expect(response).to have_http_status(422)
+    expect(parsed_response).to be_valid_against_openapi_schema('BadRequestBodyResponse')
```

Comment:
Can we add a test for the exact error message? 

There might be more reasons for a `422` (https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/283) and I'd like to confirm the resulting error message is readable.

Reply:
Adding test, I found out that the  error message contains 'User can't be blank',
that is wrong, it should be 'User can't be blank', it might be an easy fix, I probably need to tinker a bit with the `en.yml`
cc @duncanjbrown

Reply:
I added the spec for the error messages.
There was a minor glitch in the error message:

`Full name can't be blank, Email can't be blank, and User can't be blank`

It shows `User` instead of `user_id`, because to a rails conversions, probably to humanize the error message.

 ATM I fixed that with a custom error on `en.yml`:

`Full name can't be blank, Email can't be blank, and User 'user_id' can't be blank`

That better, but still not perfect (it might be enough for now).

cc @duncanjbrown

Reply:
aThis works:

```
  activemodel:
    attributes:
      vendor_api/attribution_meta:
        user_id: "user_id"
```

Result:

```
"Invalid Metadata Timestamp can't be blank, Attribution is invalid: Full name can't be blank, Email can't be blank, and user_id can't be blank"
```

Reply:
While we're here, why not go the whole way and render an error message which corresponds exactly to the input fields:

```
  activemodel:
    attributes:
      vendor_api/metadata:
        timestamp: "timestamp"
        attribution: "attribution"
      vendor_api/attribution_meta:
        user_id: "user_id"
        email: "email"
        full_name: "full_name"
```

```
 "Invalid Metadata timestamp can't be blank, attribution is invalid: full_name can't be blank, email can't be blank, and user_id can't be blank" 
```

Reply:
that's exactly what we want!
nitpicking: add a colon `:`
`Invalid Metadata: timestamp ...`

Reply:
The actual errors now are

```
[{"error"=>"ValidationError", "message"=>"timestamp can't be blank"}, {"error"=>"ValidationError", "message"=>"attribution is invalid: full_name can't be blank, email can't be blank, and user_id can't be blank"}]
```